pytorch_lightning/trainer/trainer.py

```diff
@@ -1773,6 +1774,9 @@ def _log_device_info(self) -> None:
         num_ipus = self.ipus if self.ipus is not None else 0
         rank_zero_info(f"IPU available: {_IPU_AVAILABLE}, using: {num_ipus} IPUs")
 
+        num_hpus = self.devices if self.devices is not None else 0
+        rank_zero_info(f"HPU available: {_HPU_AVAILABLE}, using: {num_hpus} HPUs")
```

Comment:
@kaushikb11 if i'm testing something on a CPU-only, I now have to see logs/warnings for GPUs, TPUs, IPUs, and now HPUs? this is super noisy for me. IMO we should only log these statements if the hardware is available.

Reply:
I agree, but while I see it a bit noisy, it's also nice to put there to showcase accelerators which Lightning supports.

Reply:
> we should only log these statements if the hardware is available.

Tbh, I feel this is a fair argument. Let's do it for this release, then we could discuss how we would go ahead with it.